haystack/components/agents/agent.py

```diff
@@ -609,13 +598,15 @@ async def run_async(  # noqa: PLR0915
             )
             counter = 0
 
-            if break_point and self._agent_name is None:
-                raise ValueError("When using breakpoints, the agent_name must be provided to save the state correctly.")
-
```

Comment:
I dropped this since we always set the agent_name to either `__component_name__` or `"isolated_agent"` so this check was never triggered

Reply:
good point